server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/PlugableDataFetcher.java

```diff
@@ -0,0 +1,36 @@
+package io.smallrye.graphql.execution.datafetcher;
+
+import graphql.GraphQLContext;
+import graphql.execution.DataFetcherResult;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import io.smallrye.graphql.bootstrap.Config;
+import io.smallrye.graphql.execution.context.SmallRyeContext;
+import io.smallrye.graphql.schema.model.Operation;
+import io.smallrye.graphql.spi.DataFetcherService;
+
+/**
+ * Delegate the actual fetching to a SPI implementation.
+ * 
+ * @author Phillip Kruger (phillip.kruger@redhat.com)
+ */
+public class PlugableDataFetcher<T> implements DataFetcher<T> {
+    private final DataFetcherService dataFetcherService;
+    private final Operation operation;
+
+    public PlugableDataFetcher(Operation operation, Config config) {
+        this.operation = operation;
+        this.dataFetcherService = DataFetcherService.getDataFetcherService(operation);
+        this.dataFetcherService.init(operation, config);
+    }
+
+    @Override
+    public T get(final DataFetchingEnvironment dfe) throws Exception {
+        SmallRyeContext.setDataFromFetcher(dfe, operation);
```

Comment:
Does that use some kind of thread local?

Reply:
or is it just a static singleton.

Reply:
Yes local thread.